tests/cypress.config.js

```diff
@@ -43,15 +42,15 @@ module.exports = defineConfig({
               if (data.includes(PROXY_WAITING_FOR_DEX_MARKER)) {
                 resolve(data.toString().split(' ')[1])
```

Comment:
This could be something like:
```suggestion
                throw WaitingForDexAuthenticationException(data.toString().split(' ')[1])
```

Reply:
Just for the sake of it: in my book, while usually the case, an exception doesn't necessarily need to indicate an error or something failing. For example, `io.EOF` being returned as an error in Go or `KeyboardInterrupt` being thrown in Python. I think using exceptions in certain cases for signaling or control flow can actually be beautiful rather than trying to differentiate between different return values. I think retries could be one of those cases.